src/main/java/uk/gov/hmcts/ccd/domain/service/search/elasticsearch/security/AuthorisedCaseSearchOperation.java

```diff
@@ -74,7 +80,13 @@ public CaseSearchResult execute(CrossCaseTypeSearchRequest searchRequest) {
 
     private CrossCaseTypeSearchRequest createAuthorisedSearchRequest(List<CaseType> authorisedCaseTypes, CrossCaseTypeSearchRequest originalSearchRequest) {
         List<String> authorisedCaseTypeIds = authorisedCaseTypes.stream().map(CaseType::getId).collect(Collectors.toList());
-        return new CrossCaseTypeSearchRequest(authorisedCaseTypeIds, originalSearchRequest.getSearchRequestJsonNode());
+
+        return new CrossCaseTypeSearchRequest.Builder()
+            .withCaseTypes(authorisedCaseTypeIds)
+            .withSearchRequest(originalSearchRequest.getSearchRequestJsonNode())
+            .withMultiCaseTypeSearch(originalSearchRequest.isMultiCaseTypeSearch())
```

Comment:
I would remove withMultiCaseTypeSearch and also the setter on the request. And just leave a getter based on collection size as we were doing before. 
Much less code

Reply:
Case types passed to the request may not be all authorised. If there are 2 case types supplied and the user is authorised to view only one. The semantics of multi case type search changes to single.